.travis.yml

```diff
@@ -1,3 +1,6 @@
+env:
+- API_REGEX='Google\.(Cloud\.)?[A-M].*'
```

Comment:
This one looks like it will capture all of them, which is probably why one of the splits is taking 2x as long. For example, in "Google.Cloud.Vision.V1", the 'V' fails to be matched by `[A-M]`, so the optional `(Cloud\.)` cannot be used, so the regex will backtrack and capture the leading 'C' from "Cloud" with the `[A-M]` pattern and then `.*` will capture "loud.Vision.V1"

It's a shame bash regex doesn't support negative lookahead assertions, or this would be rather clean:
`Google\.(Cloud\.)?[A-M](?!loud\.).*`
meaning "don't match the `[A-M]` pattern if it is followed by "loud."

It looks like this might be the next best option:
`Google\.((Cloud\.)?[ABD-M])|(C[^l]).*`
which matches 'A'-'M' (except 'C') after an optional "Cloud." OR it matches 'C' which is followed by a character that is not 'l', and then anything else.

As Ian said, "Regexs are hard :)"

Reply:
Humbug. I thought I'd checked that all locally. Will look again - thanks for the explanation.

I suspect the simplest option may be to handle non-cloud APIs separately, explicitly listing Google.LongRunning as another job for now. Then make the Cloud part mandatory for everything else.